app/models/resource.rb

```diff
@@ -75,6 +78,10 @@ def file_ext
     (Mime::Type.file_extension_of mime).parameterize
   end
 
+  def raw_url
+    'http://' + NB.host + '/' + File.join(Rails.root, 'resources', 'raw', "#{ mime == 'application/pdf' ? local_file_name : id }.#{ file_ext }")
```

Comment:
Line is too long. [144/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.